src/processor.rs

```diff
@@ -320,8 +322,8 @@ impl<'a> EventProcessor<'a> {
         transaction_context: &mut TRANSACTION_CONTEXT,
     ) -> Result<(), EventHandlerError> {
         for (event_index, event) in self.transaction.events.iter().enumerate() {
-            let handler_exists = handler_registry
-                .handler_exists(event.emitter.address(), &event.name);
+            let handler_exists =
+                handler_registry.handler_exists_for_event(event);
```

Comment:
Thought to make it clearer that it is related to an event, since this is a method on the HandlerRegistry, but I think it's fine to change it back since the parameter "event" makes that clear enough